Kubernetes/windows/debug/enablewinflags/winserver2022.yaml

```diff
@@ -0,0 +1,66 @@
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: enable-win-flags
+  labels:
+    app: enable-win-flags
+spec:
+  selector:
+    matchLabels:
+      name: enable-win-flags
+  template:
+    metadata:
+      labels:
+        name: enable-win-flags
+    spec:
+      securityContext:
+        windowsOptions:
+          hostProcess: true
+          runAsUserName: "NT AUTHORITY\\SYSTEM"
+      hostNetwork: true
+      containers:
+      - name: enable-win-flags
+        image: mcr.microsoft.com/dotnet/framework/samples:aspnetapp
+        command:
+        - powershell.exe
+        - -command
+        - |
+            function EnableWinFlags {
+
+              $count = (reg query HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Policies\Microsoft\FeatureManagement\Overrides | sls "REG_DWORD" ).Count
+              $regKeySet = (reg query HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Policies\Microsoft\FeatureManagement\Overrides | sls "3444861581" ).Count
+
+              if(($count -ge 3) -and ($regKeySet -EQ 1)) {
+                Write-Host "Win flags are already enabled."
+                return
+              }
+
+              Write-Host "Adding registry key to enable the feature"
+              reg add HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Policies\Microsoft\FeatureManagement\Overrides /v 3444861581 /t REG_DWORD /d 1 /f
+              Start-Sleep -Seconds 2
+
+              Write-Host "Updating group policy"
+              gpupdate /force
+              Start-Sleep -Seconds 2
+
+              Write-Host "Running tasks to read the changes in step 1"
+              schtasks /run /tn "Microsoft\Windows\Flighting\FeatureConfig\ReconcileFeatures"
+              Start-Sleep -Seconds 2
+
+              Write-Host "Restarting computer for changes to take effect"
+              Start-Sleep -Seconds 5
+              Restart-Computer
+            }
+
+            EnableWinFlags
+
+            While($true) {
+              Write-Host "Ready to delete host process container."
+              Start-Sleep -Seconds 60
+            }
+
+        securityContext:
+          privileged: true
+      nodeSelector:
+        kubernetes.azure.com/os-sku: Windows2022
+        kubernetes.io/hostname: aksnpwin000000
```

Comment:
Do we want to run this DaemonSet on every Node, or only on one node?
kubernetes.io/hostname: aksnpwin000000 Would mean we schedule it only on the Node called `aksnpwin000000` 
Some customers may have different name for their nodes.

If we want to schedule on only specific nodes, it might be better to use tag instead of hostname. Then they will not need to edit the manifest.

Instead we ask user to add tag to the nodes where they want fix applied using something like
`kubectl label nodes <node-1-name> winflags=3444861581`

And we replace `kubernetes.io/hostname: aksnpwin000000` with filter for our label `winflags:3444861581`

Reply:
@daschott If the user wants to update per cluster, will daemonset be the teh right approach ?

Reply:
Yes, if you want to run it on all the Windows Nodes in the cluster, just remove the `kubernetes.io/hostname: aksnpwin000000`. 
Then it will target all Windows nodes and you don't need to add any labels.

Reply:
And yes, daemonset is the right approach

Reply:
@daschott  Shall I leave the decision to the customer. Because they might have different approach. I will convey the same to them to choose what is best for them.

Reply:
sounds good